chain/chain-primitives/src/error.rs

```diff
@@ -388,3 +388,25 @@ impl From<StorageError> for Error {
         ErrorKind::StorageError(error).into()
     }
 }
+
+#[derive(Clone, Eq, PartialEq, Debug, Fail)]
+pub enum BlockKnownError {
+    KnownInHeader,
+    KnownInHead,
+    KnownInOrphan,
+    KnownInMissingChunks,
+    KnownInStore,
+}
+
+impl Display for BlockKnownError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let output = match self {
+            BlockKnownError::KnownInHeader => "already known in header",
+            BlockKnownError::KnownInHead => "already known in head",
+            BlockKnownError::KnownInOrphan => "already known in orphan",
+            BlockKnownError::KnownInMissingChunks => "already known in missing chunks",
+            BlockKnownError::KnownInStore => "already known in store",
+        };
+        Display::fmt(&output, f)
+    }
+}
```

Comment:
```suggestion
impl fmt::Display for BlockKnownError {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let output = match self {
            BlockKnownError::KnownInHeader => "already known in header",
            BlockKnownError::KnownInHead => "already known in head",
            BlockKnownError::KnownInOrphan => "already known in orphan",
            BlockKnownError::KnownInMissingChunks => "already known in missing chunks",
            BlockKnownError::KnownInStore => "already known in store",
        };
        fmt::Display::fmt(&output, f)
    }
}
```

to minimize the amount of requried imports to jsut `fmt`